src/Checkpoints.jl

```diff
@@ -25,25 +25,22 @@ __init__() = Memento.register(LOGGER)
 include("handler.jl")
 
 const CHECKPOINTS = Dict{String, Union{Nothing, Handler}}()
-@contextvar TAGS = OrderedDict{Symbol, Any}()
+@contextvar CONTEXT_TAGS::Tuple{Vararg{Pair{Symbol, Any}}} = Tuple{}()
 
 include("session.jl")
 
 """
-    with_tags(f::Function, tags::Pair...)
+    with_tags(f::Function, context_tags::Pair...)
 
-Runs the function `f` with dynamically scoped [1] `tags`. These tags are used in addition
-to the tags provided to the [`checkpoint`](@ref) function when determining where to save
-the checkpoints.
-
-Nested `with_tags` are allowed. The values of tags in the innermost `with_tags` call take
-precedence. The `tags` passed to `with_tags` should be disjoint from the tags used in the
+Creates a new context and runs the function `f` within the context. `context_tags` are used
+everywhere inside the context in addition to the tags provided directly in the
```

Comment:
````suggestion
Runs the function `f`, tagging any [`checkpoint`](@ref)s created by `f` with the `context_tags`.
This is normally used via the do-block form:
For example

```julia
with_tags(:foo=>1, :bar=>2) do
    q_out = qux()
    checkpoint("foobar", :output=q_out)
end
```
This snippted will result in   `"foobar"` checkpoint having the `foo=1` and `bar=2` tags, as will any checkpoints created by `qux`().
The context tags are[dynamically scoped](https://en.wikipedia.org/wiki/Scope_(computer_science)#Lexical_scope_vs._dynamic_scope_2) and so are remembered through function calls.
````